src/textual_serve/download_manager.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+import asyncio
+from contextlib import suppress
+from dataclasses import dataclass, field
+import logging
+from typing import AsyncGenerator, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from textual_serve.app_service import AppService
+
+log = logging.getLogger("textual-serve")
+
+DOWNLOAD_TIMEOUT = 4
+DOWNLOAD_CHUNK_SIZE = 1024 * 64  # 64 KB
+
+
+@dataclass
+class Download:
+    app_service: "AppService"
+    delivery_key: str
+    file_name: str
+    open_method: str
+    mime_type: str
+    encoding: str | None = None
+    incoming_chunks: asyncio.Queue[bytes | None] = field(default_factory=asyncio.Queue)
+
+
+class DownloadManager:
+    """Class which manages downloads for the server.
+
+    Serves as the link between the web server and app processes during downloads.
+
+    A single server has a single download manager, which manages all downloads for all
+    running app processes.
+    """
+
+    def __init__(self):
+        self._active_downloads_lock = asyncio.Lock()
```

Comment:
Could be mistake, but I don't think you will need this lock.

Everything between a `await` is essentially atomic. i.e. if you don't await anything in the locked block, then there is no need for a lock...